gamemodes/zombieplague/gamemode/extraitems/zp_armor.lua

```diff
@@ -1,6 +1,25 @@
 ExtraItem.ID = "ZPArmor"
 ExtraItem.Name = "ExtraItemArmorName"
-ExtraItem.Price = 12
+ExtraItem.Price = 10
+ExtraItem.Type = ITEM_HUMAN 
+ExtraItem.BuySounds = { "npc/combine_soldier/gear6.wav" }
 function ExtraItem:OnBuy(ply)
-	ply:SetArmor(ply:Armor() + 100)
-end
\ No newline at end of file
+	ply:SetNWInt("ZPBuyedArmor", ply:GetNWInt("ZPBuyedArmor") + 1)
+	local addArmor = ply:Armor() + 100
+	ply:SetArmor(math.Clamp(addArmor, 0, ply:GetMaxHealth()))
+end
+
+function ExtraItem:CanBuy(ply)
+	if !ply:IsHuman() then return false end
+	if !ply:Alive() then return false end
+	if ply:GetNWInt("ZPBuyedArmor") <= 1 then
+		return ply:Armor() < ply:GetMaxHealth()
```

Comment:
So the player can only buy armor to fill the amount of "max health"? Is this to prevent a player with 100000000 armor?

Reply:
It also gives some effect on balance